dataeng/resources/tableau-backup.sh

```diff
@@ -1,30 +1,39 @@
 #!/usr/bin/env bash
-set -ex
 
 # This script ssh into Tableau server. Default centos user does not have access on Tableau directories
 # so shared folder is created that can be accessible from centos user. Backups are being created using
-# tsm maintenance backup utility. This script copies backup file into s3 bucket and then remove backup
-# file from tableau server.
+# tsm maintenance backup utility. This script creates the data and config backup and copies them into s3
+# bucket and then remove backup files from tableau server.
+
+ssh $USER_NAME@$TABLEAU_SERVER_IP /bin/bash << EOF
+set -ex
+sudo su - tsm_admin
+set -ex
+
+# Data backup filename in the format of table_data_backup_2022_01_16_T23_48Z.tsbak
+data_backup_filename="tableau_data_backup_$(date -u +%Y_%m_%d_T%H_%MZ).tsbak"
+
+echo "Creating data backup"
+tsm maintenance backup --file \$data_backup_filename --multithreaded
+
+echo "Copying data backup file into s3"
+aws s3 cp /home/tableau_backups/\$data_backup_filename s3://$S3_BUCKET/\$data_backup_filename
+
+echo "Removing data backup file from local disk"
+rm /home/tableau_backups/\$data_backup_filename
 
-ssh -tt $USER_NAME@$TABLEAU_SERVER_IP -o StrictHostKeyChecking=no "S3_BUCKET=$S3_BUCKET" '
+# Config backup filename in the format of tableau_config_backup_2022_01_16_T23_48Z.json
+config_backup_filename="tableau_config_backup_$(date -u +%Y_%m_%d_T%H_%MZ).json"
 
-# It gets the current date from system in 2021.10.05 format
-DATE="$(date '+%Y.%m.%d')"
-# It adds the date as prefix on backup name. Backup runs at 11:00 UTC so it also prefixed in the name.
-BACKUPNAME="$DATE.11.00-tableau-data-backup.tsbak"
+echo "Creating config backup"
+tsm settings export -f \$config_backup_filename
 
-# Check if tableau backup file exist, exit if the file does not exist
-if [[ ! -f /home/tableau_backups/$BACKUPNAME ]]
-then
-    echo "$BACKUPNAME Backup file does not exist. It seems to be some problem generating tableau backup. Refer to Administration of Tableau wiki docs to debug Tableau Regular Backup Automation"
-    exit 1
-else
-    echo "Copying backup file into s3"
-    sudo aws s3 cp /home/tableau_backups/$BACKUPNAME s3://$S3_BUCKET/$BACKUPNAME
+echo "Copying config backup file into s3"
+aws s3 cp \$config_backup_filename s3://$S3_BUCKET/\$config_backup_filename
 
-    echo "Removing backup file from local disk"
-    sudo rm -f /home/tableau_backups/$BACKUPNAME
+echo "Removing config backup file from local disk"
+rm \$config_backup_filename
```

Comment:
Can you change this to delete _all_ backups, in case any were left over from previous runs?  If they really are 40+ GB, then we should try to prevent accidental accumulation on disk.